lib/packages/docutils/rst.nim

```diff
@@ -1891,28 +1930,36 @@ proc dirRaw(p: var RstParser): PRstNode =
   else:
     dirRawAux(p, result, rnRaw, parseSectionWrapper)
 
+proc dirUnimpl(p: var RstParser, d: string): PRstNode =
+  rstMessage(p, meInvalidDirective, d)
+  result = nil
+
+var directives = {
+  "": dirUnimpl,
+  "code": dirCodeBlock,
+  "code-block": dirCodeBlock,
+  "container": dirContainer,
+  "contents": dirContents,
+  "figure": dirFigure,
+  "image": dirImage,
+  "include": dirInclude,
+  "index": dirIndex,
+  "raw": dirRaw,
+  "title": dirTitle
+}.toTable
```

Comment:
For a Table this small Im not really sure if its worth it, linear search is pbl better

Reply:
I'm not sure. Let @narimiran decide.

I moved to the table-based implementation only because it seemed more compact and clear than previous one.

Also note that though there are only 20 directives currently, but there are more of them in the RST spec.

Reply:
I only count ten but I will make a bench and return.

Reply:
So yes its faster/uses less memory https://gist.github.com/planetis-m/fa12f6c4ff528babf9f15f3d189038de

Reply:
...and you dont need all those workarounds because you need a uniform function signature

Reply:
yes, but there is no need in enums then. A large case statement on strings works even faster https://gist.github.com/a-mr/fe3596a928df860503d7f946026d50a3

Reply:
thank you @planetis-m ,
moved to a simple case statement.